src/test/java/de/tum/in/www1/artemis/notification/SingleUserNotificationServiceTest.java

```diff
@@ -306,6 +343,92 @@
         assertThat(mimeMessageCaptor.getValue().getSubject()).isEqualTo(PLAGIARISM_CASE_VERDICT_STUDENT_TITLE);
     }
 
+    @Test
+    void testConversationNotifications_oneToOneChatCreation() {
+        singleUserNotificationService.notifyUserAboutConversationCreationOrDeletion(oneToOneChat, user, userTwo, CONVERSATION_CREATE_ONE_TO_ONE_CHAT);
+        List<Notification> capturedNotifications = notificationRepository.findAll();
+        assertThat(capturedNotifications).as("Notification should not have been saved").hasSize(0);
+        // notification should be sent
+        verify(messagingTemplate, times(1)).convertAndSend(eq("/topic/user/" + user.getId() + "/notifications"), (Object) any());
+    }
+
+    @Test
+    void testConversationNotifications_groupChatCreation() {
+        singleUserNotificationService.notifyUserAboutConversationCreationOrDeletion(groupChat, user, userTwo, CONVERSATION_CREATE_GROUP_CHAT);
+        verify(messagingTemplate, times(1)).convertAndSend(eq("/topic/user/" + user.getId() + "/notifications"), (Object) any());
+
+        singleUserNotificationService.notifyUserAboutConversationCreationOrDeletion(groupChat, userThree, userTwo, CONVERSATION_CREATE_GROUP_CHAT);
+        verify(messagingTemplate, times(1)).convertAndSend(eq("/topic/user/" + userThree.getId() + "/notifications"), (Object) any());
+
+        List<Notification> capturedNotifications = notificationRepository.findAll();
+        assertThat(capturedNotifications).as("Both notifications should have been saved").hasSize(2);
+        capturedNotifications.forEach(capturedNotification -> {
+            assertThat(capturedNotification.getTitle()).as("Title of the captured notification should be equal to the expected one")
+                    .isEqualTo(CONVERSATION_CREATE_GROUP_CHAT_TITLE);
+        });
+    }
+
+    @Test
+    void testConversationNotifications_groupChatAddUsers() {
```

Comment:
In general a lot of tests are very similar in structure. Is there a way to parameterize them?

Reply:
I thought to do so as well but i could not find a solution to be honest. IMO I would leave as is but if you might have any hints i can look into that again.

Reply:
Maybe this: https://www.baeldung.com/parameterized-tests-junit-5

Reply:
I parameterized some tests 👍